checker/src/org/checkerframework/checker/signedness/SignednessVisitor.java

```diff
@@ -23,6 +22,124 @@ public SignednessVisitor(BaseTypeChecker checker) {
         super(checker);
     }
 
+    /** @return true iff node is a mask operation (&amp; or |) */
+    private boolean isMask(Tree node) {
+        Kind kind = node.getKind();
+
+        return kind == Kind.AND || kind == Kind.OR;
+    }
+
+    /** @return true iff expr is a literal */
+    private boolean isLiteral(ExpressionTree expr) {
+        return expr instanceof LiteralTree;
+    }
+
+    /**
+     * @param obj either an Integer or a Long
+     * @return the long value of obj
+     */
+    private long getLong(Object obj) {
+        if (obj instanceof Integer) {
+            Integer intObj = (Integer) obj;
+            return intObj.longValue();
+        } else {
+            return (long) obj;
+        }
+    }
+
+    /**
+     * Given a masking operation of the form {@code x & maskLit} or {@code x | maskLit}, return true
+     * iff the masking operation results in the same output regardless of the value of the numBits
+     * most significant bits of x. This is if the numBits most significant bits of mask are 0 for
+     * AND, and 1 for OR. For example, assuming that numBits is 4, the following is true about AND
+     * and OR masks:
+     *
+     * <p>{@code x & 0x0... == 0x0...;} {@code x | 0xF... == 0xF...;}
```

Comment:
It's slightly confusing to use `x` as the variable when you use the letter x in the constant `0x0...`.
Also, x can be an arbitrary expression, but `x` suggests it's a variable; "expr" might be better.
Also, the ellipsis is different on the left and right side of the `==`, but I'm not sure exactly how to express that fact which took me a moment to realize.